STUDIO/org.openl.rules.jackson/src/org/openl/rules/serialization/ObjectMapperCache.java

```diff
@@ -0,0 +1,72 @@
+package org.openl.rules.serialization;
+
+/*-
+ * #%L
+ * OpenL - STUDIO - Jackson
+ * %%
+ * Copyright (C) 2016 - 2021 OpenL Tablets
+ * %%
+ * See the file LICENSE.txt for copying permission.
+ * #L%
+ */
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.WeakHashMap;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class ObjectMapperCache {
+
+    private static final WeakHashMap<List<Class<?>>, ObjectMapper> cache = new WeakHashMap<>();
+    private final static ReentrantReadWriteLock lock = new ReentrantReadWriteLock();
+    private static volatile ObjectMapper defaultObjectMapper;
+
+    /**
+     * ObjectMapper for set of classes
+     *
+     * @param classes set of classes
+     * @return objectMapper
+     */
+    public static ObjectMapper getObjectMapper(Class<?>[] classes) throws ClassNotFoundException {
+        List<Class<?>> key = Arrays.asList(classes);

```

Comment:
As far as I remember, GC will delete the value from the memory which is stored by this key, when the key is not in use in the application in any place.
In your implementation, an object key is created and used only here

Reply:
I'm not sure if java cares about nested links. Please correct me if I'm wrong

Reply:
Agree, this implementation can be improvement